app/views/two_factor_authentication/totp_verification/show.html.erb

```diff
@@ -6,18 +6,16 @@
 
 <%= form_tag(:login_two_factor_authenticator, method: :post, role: 'form', class: 'margin-top-4 tablet:margin-top-6') do %>
   <%= render @presenter.reauthn_hidden_field_partial %>
-  <%= label_tag 'code', t('forms.two_factor.code'), class: 'block bold' %>
+  <%= label_tag :code, t('forms.two_factor.code'), class: 'block bold' %>
   <div class="col-12 sm-col-5 margin-bottom-1 tablet:margin-bottom-0 sm-mr-20p inline-block">
-    <%= text_field_tag :code, '',
-                       value: @code,
-                       required: true,
-                       aria: { invalid: false, describedby: 'code-instructs' },
-                       autofocus: true,
-                       pattern: '[0-9]*',
-                       class: 'col-12 field monospace mfa',
-                       type: 'tel',
-                       maxlength: TwoFactorAuthenticatable::OTP_LENGTH,
-                       autocomplete: 'one-time-code' %>
+    <%= render(
+      'shared/one_time_code_input',
```

Comment:
If this is for the authentication app codes, maybe we don't need or want SMS code auto-fill. Then again, it shouldn't hurt to include, and could at least help toward ensuring consistent "code entry" fields.

Reply:
yeah definitely don't want to autofill for 2FA... maybe we add an `autocomplete: :none` variable or something?

Reply:
I'm on the fence, leaning toward one or the other of:

1. Revert this, and maybe go a step further to remove the `autocomplete` attribute that had been assigned
2. Treat the app setup as a "one-time-code", but adapt to allow customization of transport, where we avoid SMS for this field. The spec does mention [possible future transports](https://wicg.github.io/web-otp/#transports), which could maybe include auth app?

Pseudo-code:

```erb
<%= tag.input data: { transport: local_assigns[:transport] || 'sms' }, class: 'one-time-code-input' %>
```

```js
const input = document.querySelector('.one-time-code-input');
if (input?.dataset.transport) {
  const { code } = await navigator.credentials.get({ otp: { transport: [input.dataset.transport] }});
  // ...
}
```

Reply:
Yeah it's good to know the spec is flexible but I wouldn't want to leave in something speculative like that in our code?

If the goal is visual consistency, would love to find a way to keep things consistent via partial, but maybe disable some of the extra behavior attrs

One case I think we should avoid is:
1. I click SMS, go to SMS page
2. it takes too long, so I go switch to TOTP
3. but then the SMS arrives

In that case, I wouldn't want the SMS prefilled in the TOTP

Reply:
Yeah, I should have been more clear, but my thinking with what the specification mentions w.r.t. future supports isn't so much something where I'd build in support for those hypothetical methods now, but more as affirmation that reusing the partial for visual consistency wouldn't be inappropriate, especially if we were to continue calling it a "one time code input" for the TOTP setup screens.

As proposed, the partial would only currently support two states: SMS as the transport method (`transport: 'sms'`), or disabling the behavior outright (`transport: nil`). If and when future transports are added, we could choose to build upon this if we want to support those.

See: 3d255654f32cdc32973fb390801ee9fa97bddf15, a811bdddf8896c9d15102276c63694d2c6df1b2b

Reply:
ah ok great, thanks for clarifying